search/search_repository.go

```diff
@@ -0,0 +1,88 @@
+package search
+
+import (
+	"golang.org/x/net/context"
+
+	"log"
+	"strconv"
+
+	"github.com/almighty/almighty-core/app"
+	"github.com/almighty/almighty-core/models"
+)
+
+const (
+	fulltextPsqlQuery = `select * from work_items WHERE to_tsvector('english', id::text || ' ' || fields::text) @@ to_tsquery($1)`
+)
+
+// GormSearchRepository provides a Gorm based repository
+type GormSearchRepository struct {
+	ts  *models.GormTransactionSupport
+	wir *models.GormWorkItemTypeRepository
+}
+
+// NewGormSearchRepository creates a new search repository
+func NewGormSearchRepository(ts *models.GormTransactionSupport, wir *models.GormWorkItemTypeRepository) *GormSearchRepository {
+	return &GormSearchRepository{ts, wir}
+}
+
+func generateSearchQuery(q string) (string, error) {
+	return q, nil
+}
+
+func convertFromModel(wiType models.WorkItemType, workItem models.WorkItem) (*app.WorkItem, error) {
+	result := app.WorkItem{
+		ID:      strconv.FormatUint(workItem.ID, 10),
+		Type:    workItem.Type,
+		Version: workItem.Version,
+		Fields:  map[string]interface{}{}}
+
+	for name, field := range wiType.Fields {
+		var err error
+		result.Fields[name], err = field.ConvertFromModel(name, workItem.Fields[name])
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &result, nil
+}
+
+func (r *GormSearchRepository) loadTypeFromDB(ctx context.Context, name string) (*models.WorkItemType, error) {
```

Comment:
My comment was aimed at why we're not adding search functionality to the work item repository instead of implementing it here. That is a different discussion. 
Since we are accessing the database directly in this package anyway, it doesn't really make a difference: there is not much smarts in the loadTypes... function anyway.